### PR TITLE
Full changes to pipeline to dehost Paired Illumina data

### PR DESCRIPTION
**Pull request by DarianHole**
# Initial PR for Illumina data

- Map to human and viral reference with bwa
- Keep only paired reads that are in the viral reference and remove any reads that are found in the human reference
- Read quality options for what is tolerated as a good human or viral read (strict by default)
- Space and setup for nanopore data along with single-end fastq reads

### CODE REVIEW

File: conf/illumina.config [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    min_length = 50
+
+    // Maximum fastq read length
+    max_length = 300
+
+    // Single-end reads as input
+    single_end = false
+
+    // illumina fastq search path
+    illuminaSuffixes = ['*_R{1,2}_001', '*_R{1,2}', '*_{1,2}' ]
```
Comment (Takadonet) [inline conf/illumina.config]: add in pair1 and pair2 as well

File: bin/dehost.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    samfile = pysam.AlignmentFile(samfile_in, "r")
+    header = samfile.header
+
+    for read in samfile.fetch():
+        if read.query_name in removal_read_ids:
+            pass
+
+        elif read.mapping_quality >= input_mapping_quality:
+            reads_found.append(read)
```
Comment (Takadonet) [inline bin/dehost.py]: else statement where read does not match human OR virus mapping quality, count that as well and report back.

**Review by Takadonet — CHANGES_REQUESTED** [review]

File: conf/illumina.config (hunk 1 above)
Comment (DarianHole) [reply conf/illumina.config]: also add gzip and gz

File: bin/dehost.py (hunk 2 above)
Comment (DarianHole) [reply bin/dehost.py]: Potentially also use counts with flagstats outputs to generate output metrics in one file

Comment (DarianHole) [reply bin/dehost.py]: Good for the moment, will have to re-visit for flagstats stuff (as its accounted for, just not the most accessible needing to read a different file)

File: conf/illumina.config (hunk 1 above)
Comment (DarianHole) [reply conf/illumina.config]: Still need to figure out why adding gz to the end of the file finder messes it up...

**Review by Takadonet — APPROVED** [review]

**Pull request merged**